admin/app/components/solidus_admin/ui/table/component.html.erb

```diff
@@ -31,7 +31,7 @@
               font-[600]
               line-[120%]
             "
-          ><%= render_header_cell(column.header) %></th>
+          ><%= render_header_cell(column.header) %></<%= header_cell_tag(column) %>>
```

Comment:
Great catch!

If we move the whole tag generation inside the `render_header_cell` helper we can make it work even when a header will return blank content through a block or a method call (Symbol).

I have a preference for seeing all the classes in the template file, but in this case would be fine to move them to the class.

Reply:
Thanks for your suggestion! I didn't want to refactor the work for the table component further for such a small thing, as we have both the `render_header_cell` & `render_data_cell` going through a common `render_cell` method that doesn't deal with HTML tags. But I'm entirely ok with changing it now. Please, can you take a look at the last commit I pushed? I will squash it with the relevant previous commit if you like it. Otherwise, I can remove it and leave it as the initial PR.

Reply:
I had a look and it's way better, I think there's still something that can be improved on the parity between `render_header_cell` and `render_data_cell`: 
- they should both render their `td` or `th` tag
- `render_header_cell` lost the ability to render components, previously done by `render_cell`
- you can leverage `content_tag` and pass the tag from outside

```rb
def render_header_cell(cell)
  cell =
    case cell
    when Symbol
      @model_class.human_attribute_name(cell)
    when Proc
      cell.call
    end
  cell = cell.render_in(self) if cell.respond_to?(:render_in)
  tag_name = cell.blank? ? :td : :th

  content_tag(tag_name, cell, class: %{...})
end
```

The same pattern (rendering the `td` tag) should be applied to data cells, so they look even ⚖️

Reply:
Oh, all right! Please, see my last commit and I'll squash everything together after your :heavy_check_mark: